ext/src/ruby_api/table.rs

```diff
@@ -0,0 +1,166 @@
+use super::{
+    convert::{ToRubyValue, ToWasmVal},
+    root,
+    store::{Store, StoreContextValue},
+    table_type::TableType,
+};
+use crate::{error, helpers::WrappedStruct};
+use magnus::{
+    function, memoize, method, r_typed_data::DataTypeBuilder, DataTypeFunctions, Error,
+    Module as _, Object, RClass, TypedData, Value, QNIL,
+};
+use wasmtime::Table as TableImpl;
+
+/// @yard
+/// @rename Wasmtime::Table
+/// Represents a WebAssembly table.
+/// @see https://docs.rs/wasmtime/latest/wasmtime/struct.Table.html Wasmtime's Rust doc
+#[derive(Debug)]
+pub struct Table<'a> {
+    store: StoreContextValue<'a>,
```

Comment:
These lifetimes seem off too me... can table outlive a store?

Reply:
The table marks the store, therefore the table lives at least as long as the store -- right?

Do you have an example where you think that'd be problematic?